components/dashboard/InsightsPanel.tsx

```
@@ -203,9 +203,17 @@ function InsightsPanelContent({
           if (target.trim().length === 0) {
             throw new Error('Execute target cannot be empty')
           }
-          // Simulate async operation
-          await new Promise(resolve => setTimeout(resolve, 1000))
-          // Custom action not implemented - this would be handled by the application
+          // Execute action via action handler service
+          const { actionHandler } = await import('@/lib/action-handler')
+          const result = await actionHandler.executeAction(action, {
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)